running-code/execute-script.md

```diff
@@ -0,0 +1,216 @@
+---
+jupytext:
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.16.4
+kernelspec:
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
+---
+
+(execute-script)=
+# How to Execute a Python Script
+
+There are two primary ways to execute a Python script.
+
+1. You can call Python from your shell and pass in the script path
+2. You can call your script directly as a command from your shell
+
+(execute-script-pass-to-python)=
+## 1. Pass your script to the Python command
+
+The first way to execute a Python script is to use the `python` command. The `python` command takes the name of a file and executes it from your shell, like this:
+
+```bash
+python my_program.py
+```
+
+When Python reads a file using this approach, it executes each line of code from top to bottom. Lines within functions
+or classes are not executed, unless that function is called or class is created outside of any other function or class definition.
+This is similar, but not identical, to the behavior of copying this file and pasting it line-by-line into an interactive
+Python shell (or a Jupyter notebook cell).
+
+```python
+def report_error():
+  print("An error has occured")
+
+print("\N{Sparkles} Hello from Python \N{Sparkles}")
+```
+
+Note that only one line is printed when this script is run. The `report_error` function only runs if you call it directly in the file.
+
+```bash
+python my_program.py
+# ✨ Hello from Python ✨
+```
+
+(execute-script-launch-command)=
+## 2. Associate the script with a launch command
+
+The other way a Python script may be executed is to associate the file with a launch command. The way in which
+this association is done depends on what operating system you are running.
+
+### Executing Python scripts on macOS / Linux / Non-Windows
+
+On Linux or Mac systems, the Python file can itself be turned into a command. By adding a [shebang](https://en.wikipedia.org/wiki/Shebang_(Unix))
+as the first line in any Python file, and by giving the file [executable permissions](https://docs.python.org/3/using/unix.html#miscellaneous) the
+file can be directly invoked without a `python` command.
```

Comment:
maybe something also like "you don't even need to give your python script a `.py` extension if you don't want to - to make it work like any other cli command you can call from anywhere, you could name it just "my_program" and symlink it into a directory on your $PATH" but idk if that's overkill. it would give us some continuity with the scripts section of the next lesson and also let us tie into learning objectives: "why do we want to do this at all again? oh yeah to make it easier to run our code"

Reply:
aha i see you discuss this below, nvm :)